src/libs/antares/study/parts/renewable/cluster.cpp

```diff
@@ -0,0 +1,805 @@
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include <yuni/yuni.h>
+#include <yuni/io/file.h>
+#include <yuni/core/math.h>
+#include <cassert>
+#include "../../study.h"
+#include "../../memory-usage.h"
+#include "cluster.h"
+#include "../../../inifile.h"
+#include "../../../logs.h"
+#include "../../../utils.h"
+
+using namespace Yuni;
+using namespace Antares;
+
+#define SEP IO::Separator
+
+namespace Antares
+{
+namespace Data
+{
+static bool RenewableClusterLoadFromProperty(RenewableCluster& cluster, const IniFile::Property* p)
+{
+    if (p->key.empty())
+        return false;
+
+    if (p->key == "group")
+    {
+        cluster.group(p->value);
+        return true;
+    }
+
+    if (p->key == "name")
+        return true;
+
+    if (p->key == "enabled")
+        return p->value.to<bool>(cluster.enabled);
+
+    // The property is unknown
+    return false;
+}
+
+static bool RenewableClusterLoadFromSection(const AnyString& filename,
+                                            RenewableCluster& cluster,
+                                            const IniFile::Section& section)
+{
+    if (section.name.empty())
+        return false;
+
+    cluster.name(section.name);
+
+    if (section.firstProperty)
+    {
+        // Browse all properties
+        for (auto* property = section.firstProperty; property; property = property->next)
+        {
+            if (property->key.empty())
+            {
+                logs.warning() << '`' << filename << "`: `" << section.name
+                               << "`: Invalid key/value";
+                continue;
+            }
+            if (not RenewableClusterLoadFromProperty(cluster, property))
+            {
+                logs.warning() << '`' << filename << "`: `" << section.name << "`/`"
+                               << property->value << "`: The property is unknown and ignored";
+            }
+        }
+        // update the minUpDownTime
+    }
+    return true;
+}
+
+Data::RenewableCluster::RenewableCluster(Area* parent, uint /*nbParallelYears*/) :
+ groupID(renewableOther1),
+ index(0),
+ areaWideIndex((uint)-1),
+ parentArea(parent),
+ enabled(true),
+ nominalCapacity(0.),
+ series(nullptr)
+{
+    // assert
+    assert(parent and "A parent for a renewable dispatchable cluster can not be null");
+}
+
+Data::RenewableCluster::RenewableCluster(Area* parent) :
+ groupID(renewableOther1),
+ index(0),
+ areaWideIndex((uint)-1),
+ parentArea(parent),
+ enabled(true),
+ nominalCapacity(0.),
+ series(nullptr)
+{
+    // assert
+    assert(parent and "A parent for a renewable dispatchable cluster can not be null");
+}
+
+Data::RenewableCluster::~RenewableCluster()
+{
+    delete series;
+}
+
+#ifdef ANTARES_SWAP_SUPPORT
```

Comment:
flush should be removed as well as its call

Reply:
flush triggers some actions on the TS in the swap flavor. I insist it should be kept.